src/lib/components/workflow/workflow-counts.svelte

```diff
@@ -0,0 +1,88 @@
+<script lang="ts">
+  import { page } from '$app/stores';
+  import type { WorkflowFilter } from '$lib/models/workflow-filters';
+  import { workflowStatuses } from '$lib/models/workflow-status';
+  import { workflowFilters } from '$lib/stores/filters';
+  import { isStatusFilter } from '$lib/utilities/query/filter-search';
+  import { toListWorkflowQueryFromFilters } from '$lib/utilities/query/filter-workflow-query';
+  import { combineFilters } from '$lib/utilities/query/to-list-workflow-filters';
+  import { updateQueryParameters } from '$lib/utilities/update-query-parameters';
+  import WorkflowCount from './workflow-count.svelte';
+
+  $: statusFilters = $workflowFilters.filter((filter) =>
+    isStatusFilter(filter.attribute),
+  );
+
+  function mapStatusToFilter(value: string) {
+    return {
+      attribute: 'ExecutionStatus',
+      value,
+      operator: '',
+      parenthesis: '',
+      conditional: '=',
+    };
+  }
+
+  function mapStatusesToFilters(filters: WorkflowFilter[]) {
+    if (filters.length === 1) {
+      return [mapStatusToFilter(filters[0].value)];
+    } else {
+      return filters.map((filter, i) => {
+        const operator = i === filters.length - 1 ? '' : 'OR';
+        return {
+          ...filter,
+          operator,
+        };
+      });
+    }
+  }
+
+  const onStatusClick = (status: string) => {
+    if (statusFilters.find((s) => s.value === status)) {
+      const nonStatusFilters = $workflowFilters.filter(
+        (f) => !isStatusFilter(f.attribute),
+      );
+      $workflowFilters = [
+        ...nonStatusFilters,
+        ...mapStatusesToFilters(
+          statusFilters.filter((s) => s.value !== status),
+        ),
+      ];
+    } else {
+      if (!statusFilters.length) {
+        $workflowFilters = [...$workflowFilters, mapStatusToFilter(status)];
+      } else {
+        const nonStatusFilters = $workflowFilters.filter(
+          (f) => !isStatusFilter(f.attribute),
+        );
+        $workflowFilters = [
+          ...nonStatusFilters,
+          ...mapStatusesToFilters([
+            ...statusFilters,
+            mapStatusToFilter(status),
+          ]),
+        ];
+      }
+    }
+
+    const searchQuery = toListWorkflowQueryFromFilters(
+      combineFilters($workflowFilters),
+    );
+    updateQueryParameters({
+      url: $page.url,
+      parameter: 'query',
+      value: searchQuery,
+      allowEmpty: true,
+    });
+  };
+</script>
+
+<div class="flex gap-2 lg:gap-4 flex-wrap">
```

Comment:
Only thought it that maybe we want to keep the `gap-4` here on smaller screens 
```suggestion
<div class="flex gap-4 flex-wrap">
```
so that there's still some space on hover
| gap-2 | gap-4|
|--|--|
|<img width="410" alt="Screenshot 2023-08-09 at 10 13 08 AM" src="https://github.com/temporalio/ui/assets/15069288/cfac1f5c-33ef-422e-bebc-af3999799370">|<img width="462" alt="Screenshot 2023-08-09 at 10 13 56 AM" src="https://github.com/temporalio/ui/assets/15069288/e7033d72-7a80-4dda-8d5e-08e1907af9b1">|

Reply:
I'm gonna make that box-shadow not so aggressive on sm screens, I like the smaller gap.